packages/@aws-cdk/aws-secretsmanager/lib/secret.ts

```diff
@@ -0,0 +1,259 @@
+   *
+   * @default the secret value specified in ``secretString`` is used.
+   */
+  generateSecretString?: SecretStringGenerator;
+
+  /**
+   * A name for the secret. Note that deleting secrets from SecretsManager does not happen immediately, but after a 7 to
+   * 30 days blackout period. During that period, it is not possible to create another secret that shares the same name.
+   *
+   * @default a name is generated by CloudFormation.
+   */
+  name?: string;
+}
+
+/**
+ * Attributes required to import an existing secret into the Stack.
+ */
+export interface SecretImportProps {
+  /**
+   * The encryption key that is used to encrypt the secret, unless the default SecretsManager key is used.
+   */
+  encryptionKey?: kms.IEncryptionKey;
+
+  /**
+   * The ARN of the secret in SecretsManager.
+   */
+  secretArn: string;
+}
+
+/**
+ * The common behavior of Secrets. Users should not use this class directly, and instead use ``Secret``.
+ */
+export abstract class SecretBase extends cdk.Construct implements ISecret {
+  public abstract readonly encryptionKey?: kms.IEncryptionKey;
+  public abstract readonly secretArn: string;
+
+  public export(): SecretImportProps {
+    return {
+      encryptionKey: this.encryptionKey,
+      secretArn: this.secretArn,
+    };
+  }
+
+  public grantRead(grantee: iam.IPrincipal, versionStages?: string[]): void {
+    const statement = new iam.PolicyStatement()
+      .allow()
+      .addAction('secretsmanager:GetSecretValue')
+      .addResource(this.secretArn);
+    if (versionStages != null) {
+      statement.addCondition('ForAnyValue:StringEquals', {
+        'secretsmanager:VersionStage': versionStages
+      });
+    }
+    grantee.addToPolicy(statement);
+
+    if (this.encryptionKey) {
+      this.encryptionKey.addToResourcePolicy(new iam.PolicyStatement()
+        .allow()
+        .addPrincipal(grantee.principal)
+        .addAction('kms:Decrypt')
+        .addAllResources()
+        .addCondition('StringEquals', {
+          'kms:ViaService': `secretsmanager.${cdk.Stack.find(this).region}.amazonaws.com`
+        }));
+    }
+  }
+}
+
+/**
+ * Creates a new secret in AWS SecretsManager.
+ */
+export class Secret extends SecretBase {
+  /**
+   * Import an existing secret into the Stack.
+   *
+   * @param scope the scope of the import.
+   * @param id    the ID of the imported Secret in the construct tree.
+   * @param props the attributes of the imported secret.
+   */
+  public static import(scope: cdk.Construct, id: string, props: SecretImportProps): ISecret {
+    return new ImportedSecret(scope, id, props);
+  }
+
+  public readonly encryptionKey?: kms.IEncryptionKey;
+  public readonly secretArn: string;
+
+  constructor(scope: cdk.Construct, id: string, props: SecretProps) {
+    super(scope, id);
+
+    if ((props.secretString == null) === (props.generateSecretString == null)) {
```

Comment:
not sure we should support secretString at all

Reply:
I know it's more opinionated than we are usually, but if people really want to check in their secrets into source control they should be more intentional about it. As far as I am concerned, they can use `propertyOverride`

Reply:
Okay - I'll remove support for `secretString` entirely and add some info on the "README" that if you intend on "manually" feeding the secret value, then the recommended way is to manually create the secret, and `Secret.import` it in.

Reply:
Yey!